packages/web3-eth/types/index.d.ts

```diff
@@ -0,0 +1,268 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+import * as net from 'net';
+import {AbstractProviderAdapter, provider, BatchRequest} from 'web3-providers';
+import {AbstractWeb3Module, Web3ModuleOptions, Providers} from 'web3-core';
+import {Contract} from 'web3-eth-contract';
+import {Iban} from 'web3-eth-iban';
+import {Accounts} from 'web3-eth-accounts';
+import {AbiCoder} from 'web3-eth-abi';
+import {Network} from 'web3-net';
+
+export class Eth extends AbstractWeb3Module {
+    constructor(
+        provider: AbstractProviderAdapter | provider,
+        options?: Web3ModuleOptions
+    );
+    Contract: Contract;
+    Iban: Iban;
+    personal: Personal; // change one personal types are written
+    accounts: Accounts;
+    ens: any; // change once ens types as written
+    abi: AbiCoder;
+    net: Network;
+    clearSubscriptions(): boolean
+    subscribe(type: 'logs', options?: Logs): Promise<Subscribe<Log>>;
+    subscribe(type: 'logs', callback?: (error: Error, result: Subscribe<Log>) => void): Promise<Subscribe<Log>>
+    subscribe(type: 'logs', options?: Logs, callback?: (error: Error, result: Subscribe<Log>) => void): Promise<Subscribe<Log>>;
+    subscribe(type: 'syncing', callback?: (error: Error, result: Subscribe<any>) => void): Promise<Subscribe<any>>
+    subscribe(type: 'newBlockHeaders', callback?: (error: Error, result: Subscribe<BlockHeader>) => void): Promise<Subscribe<BlockHeader>>
+    subscribe(type: 'pendingTransactions', callback?: (error: Error, result: Subscribe<Transaction>) => void): Promise<Subscribe<Transaction>>
+    setProvider(provider: AbstractProviderAdapter | provider, net?: net.Server): boolean;
+    readonly providers: Providers;
+    readonly givenProvider: AbstractProviderAdapter | provider | null;
+    BatchRequest(): BatchRequest;
+    extend(methods: Methods): any;
```

Comment:
the method extend does no longer exists.

Reply:
removed 👍

Reply:
removed and removed docs as well